src/App.jsx

```diff
@@ -8,36 +8,88 @@ import {
 	mockedAuthorsList,
 	// mockedCoursesListEmpty,
 } from './constants';
+import { Routes, Route, Navigate, useNavigate } from 'react-router-dom';
+import Registration from './components/Registration/Registration';
+import Login from './components/Login/Login';
+import CreateCourse from './components/CreateCourse/CreateCourse';
+import PropTypes from 'prop-types';
 
 function App() {
-	const [selectedItemUuid, setSelectedItemUuid] = useState('');
-	const [showInfo, setShowInfo] = useState(false);
+	const [coursesList, setCoursesList] = useState([...mockedCoursesList]);
 
+	const navigate = useNavigate();
 	const handleInfoClick = (course) => {
-		setSelectedItemUuid(course);
-		setShowInfo(true);
+		navigate(`/courses/${course.id}`);
 	};
 
 	const handleBackClick = () => {
-		setSelectedItemUuid('');
-		setShowInfo(false);
+		navigate(`/courses`);
 	};
 
+	const handleAddCourse = (course) => {
+		setCoursesList([course, ...coursesList]);
+		navigate(`/courses`);
+	};
+
+	const userToken = localStorage.getItem('userToken');
+
 	return (
 		<div>
 			<Header />
-			{!showInfo && (
-				<Courses
-					coursesList={mockedCoursesList}
-					authorsList={mockedAuthorsList}
-					onInfoClick={handleInfoClick}
+			<Routes>
+				<Route
+					path='/'
+					element={userToken ? <Navigate to='/courses' /> : <Login />}
```

Comment:
You do not need to have here condition, usually it's navigation to some page or loading LandingComponent